### PR TITLE
Add "Did You Mean This" Extension

### PR DESCRIPTION
**Pull request by christopher-o-toole**
New extension (in preview) which uses the Aladdin service to generate and show failure recovery recommendations on command failure. This extension makes use of the hook added to `parser.py` in CLI version `2.4.0`. For more information, see [this CLI PR](https://github.com/Azure/azure-cli/pull/12889).

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [X] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ai

File: src/ai-did-you-mean-this/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[bdist_wheel]
+universal=1
```
Comment (fengzhou-msft) [inline src/ai-did-you-mean-this/setup.cfg:2]: These 2 lines can be removed as we dropped support for python 2.

Comment (christopher-o-toole) [reply src/ai-did-you-mean-this/setup.cfg:2]: Removed them.

Comment (fengzhou-msft) [reply src/ai-did-you-mean-this/setup.cfg:2]: It seems you removed the one in ai-examples/setup.cfg

Comment (christopher-o-toole) [reply src/ai-did-you-mean-this/setup.cfg:2]: Woops. I'll fix that soon.

Comment (christopher-o-toole) [reply src/ai-did-you-mean-this/setup.cfg]: Fixed.

File: src/ai-did-you-mean-this/azext_ai_did_you_mean_this/commands.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# pylint: disable=line-too-long
+
+
+def load_command_table(self, _):
+
+    with self.command_group('ai-did-you-mean-this') as g:
+        g.custom_command('version', 'show_extension_version')
+
+    with self.command_group('ai-did-you-mean-this', is_preview=True):
+        pass
```
Comment (jiasli) [inline src/ai-did-you-mean-this/azext_ai_did_you_mean_this/commands.py:15]: Maybe combine them? 
```suggestion
    with self.command_group('ai-did-you-mean-this', is_preview=True) as g:
        g.custom_command('version', 'show_extension_version')
```

File: src/ai-did-you-mean-this/README.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Microsoft Azure CLI 'AI Did You Mean This' Extension
+==========================================
+
+Improve user experience by suggesting recovery options for common CLI failures.
+
+This extension extends the default error handling behavior to include recommendations for recovery. Recommendations are based on how other users were successful after they encountered the same failure.
```
Comment (jiasli) [inline src/ai-did-you-mean-this/README.rst:6]: How about adding some instructions and usage scenarios? Like what kind of error can trigger this extension.

File: src/ai-did-you-mean-this/azext_ai_did_you_mean_this/data/top_3_recommendations.json [hunk 4]
```diff
@@ -0,0 +1,11140 @@
...
+        "ad group owner": {
+            "": [
+                {
+                    "SuccessCommand": "ad group show",
+                    "SuccessCommand_Parameters": "--group",
+                    "SuccessCommand_ArgumentPlaceholders": "MyGroupDisplayName",
+                    "NumberOfPairs": 16431
+                }
+            ]
+        },
```
Comment (fengzhou-msft) [inline src/ai-did-you-mean-this/azext_ai_did_you_mean_this/data/top_3_recommendations.json:12]: This is file could be used when Aladdin service not available? But not used now?

File: src/ai-did-you-mean-this/azext_ai_did_you_mean_this/_style.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    return msg
+
+
+def should_enable_styling():
+    try:
+        if sys.stdout.isatty():
+            return True
+    except AttributeError:
+        pass
+    return False
```
Comment (fengzhou-msft) [inline src/ai-did-you-mean-this/azext_ai_did_you_mean_this/_style.py:26]: Please also honor the no_color setting: https://github.com/Azure/azure-cli/pull/12601

**Pull request closed without merging**